v2/tests/log_context_property_type_ascii_char_ptr_ut/log_context_property_type_ascii_char_ptr_ut.c

```diff
@@ -0,0 +1,530 @@
+{
+    expected_call_count = 0;
+    actual_call_count = 0;
+    actual_and_expected_match = true;
+}
+
+static void setup_expected_snprintf_call(void)
+{
+    expected_calls[expected_call_count].mock_call_type = MOCK_CALL_TYPE_snprintf;
+    expected_calls[expected_call_count].u.snprintf_call.override_result = false;
+    expected_call_count++;
+}
+
+static void setup_expected_vsprintf_call(void)
+{
+    expected_calls[expected_call_count].mock_call_type = MOCK_CALL_TYPE_vsprintf;
+    expected_calls[expected_call_count].u.vsprintf_call.override_result = false;
+    expected_call_count++;
+}
+
+static void setup_expected_vsnprintf_call(void)
+{
+    expected_calls[expected_call_count].mock_call_type = MOCK_CALL_TYPE_vsnprintf;
+    expected_calls[expected_call_count].u.vsnprintf_call.override_result = false;
+    expected_call_count++;
+}
+
+#define POOR_MANS_ASSERT(cond) \
+    if (!(cond)) \
+    { \
+        (void)printf("%s:%d test failed\r\n", __FUNCTION__, __LINE__); \
+        asserts_failed++; \
+    } \
+
+/* LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string */
+
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_001: [ If `property_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` shall fail and return a negative value. ]*/
+static void ascii_char_ptr_to_string_with_NULL_property_value_fails(void)
+{
+    // arrange
+    char buffer[128];
+    setup_mocks();
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string(NULL, buffer, sizeof(buffer));
+
+    // assert
+    POOR_MANS_ASSERT(result < 0);
+    POOR_MANS_ASSERT(actual_and_expected_match);
+    POOR_MANS_ASSERT(actual_call_count == expected_call_count);
+}
+
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_017: [ If `buffer` is `NULL` and `buffer_length` is greater than 0, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` shall fail and return a negative value. ]*/
+static void ascii_char_ptr_to_string_with_NULL_buffer_and_non_zero_buffer_length_fails(void)
+{
+    // arrange
+    setup_mocks();
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string((void*)"cucu", NULL, 1);
+
+    // assert
+    POOR_MANS_ASSERT(result < 0);
+    POOR_MANS_ASSERT(actual_and_expected_match);
+    POOR_MANS_ASSERT(actual_call_count == expected_call_count);
+}
+
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_TYPE_ASCII_CHAR_PTR_01_002: [ If `buffer` is `NULL` and `buffer_length` is 0, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string` shall return the length of the string pointed to by `property_value`. ]*/
+static void ascii_char_ptr_to_string_with_NULL_buffer_and_zero_buffer_length_returns_string_length(void)
+{
+    // arrange
+    setup_mocks();
+    setup_expected_snprintf_call();
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(ascii_char_ptr).to_string((void*)"cucu", NULL, 0);
+
+    // assert
+    POOR_MANS_ASSERT(result == 4);
```

Comment:
>4 [](http://example.com/codeflow?start=31&length=1)

strlen("cucu") is nicer :).

That would eliminate the assumption that "cucu"'s length is 4

Reply:
Changed